### PR TITLE
docs: add Codex cloud agents guide

### PR DESCRIPTION
**Pull request by owainlewis**
## Summary

- explain when to use Codex cloud and how environments work
- document the cloud CLI lifecycle, follow-ups, local application, and PR creation
- add Python examples for cloud CLI orchestration and the local Codex SDK
- describe a durable Machinist integration workflow and its asynchronous boundary

## Verification

- `go test ./...`
- `git diff --check`
- verified the documented CLI commands with Codex CLI 0.150.1
- verified the `codex cloud list --json` fields against a completed cloud task

### CODE REVIEW

File: docs/codex-cloud.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+uses the connected GitHub repository and the task's branch to create the pull request.
+
+For a terminal-controlled workflow, apply and verify the diff before pushing it yourself:
+
+```sh
+git switch -c codex/fix-control-plane-url
+codex cloud apply <TASK_ID>
+go test ./...
+git diff --check
+git add -A
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply docs/codex-cloud.md:103]: Fixed in a3a922d. The terminal PR workflow now exits unless  is empty before creating the branch, so unrelated tracked or untracked changes cannot be staged by .

File: docs/codex-cloud.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    "Fix the bug, add a regression test, and run the relevant checks",
+)
+task_id = urlparse(task_url).path.rsplit("/", 1)[-1]
+
+while True:
+    payload = json.loads(run("list", "--json", "--limit", "20"))
+    task = next(item for item in payload["tasks"] if item["id"] == task_id)
+    status = task["status"].upper()
+
+    if status in {"READY", "FAILED", "CANCELLED"}:
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply docs/codex-cloud.md]: Fixed in a5a8ad7. The polling loop now treats the CLI's , , and  values as terminal and raises for every non- outcome.

File: docs/codex-cloud.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    "exec",
+    "--env", "<ENV_ID>",
+    "--branch", "main",
+    "Fix the bug, add a regression test, and run the relevant checks",
+)
+task_id = urlparse(task_url).path.rsplit("/", 1)[-1]
+
+while True:
+    payload = json.loads(run("list", "--json", "--limit", "20"))
+    task = next(item for item in payload["tasks"] if item["id"] == task_id)
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply docs/codex-cloud.md]: Fixed in cbce750.  now follows every returned cursor and treats a task that is not listed yet as retryable, avoiding both first-page truncation and eventual-consistency failures.

File: docs/codex-cloud.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        args = ["list", "--json", "--limit", "20"]
+        if cursor:
+            args.extend(["--cursor", cursor])
+
+        payload = json.loads(run(*args))
+        for task in payload["tasks"]:
+            if task["id"] == task_id:
+                return task
+
+        cursor = payload.get("cursor")
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply docs/codex-cloud.md:151]: No change. Verified against Codex CLI 0.150.1 by running codex cloud list --json on a completed task; the top-level output fields are tasks and cursor, with cursor set to null on the last page. The documented code matches that live schema. The guide also says to pin and test the CLI integration boundary.

File: docs/codex-cloud.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    task = find_task(task_id)
+    status = task["status"].upper() if task else "NOT_LISTED"
+
+    if status in {"READY", "APPLIED", "ERROR"}:
+        break
+
+    time.sleep(10)
+
+if status != "READY":
+    raise RuntimeError(f"Codex cloud task ended with {status}")
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply docs/codex-cloud.md]: Fixed in aa992d8. APPLIED is now accepted alongside READY as a successful terminal outcome, while ERROR still raises RuntimeError.

**Comment by owainlewis** [comment]
Triage: keeping open. This guide remains useful, but the two unresolved publishing-example findings are valid. Create the local branch from the cloud task base explicitly, and stop dependent publishing commands when apply or verification fails. The guide is not obsolete; it needs these corrections before merge.
